pkg/balancer/least_connections.go

```diff
@@ -0,0 +1,129 @@
+package balancer
+
+import (
+	"math"
+	"net"
+	"sync"
+	"time"
+
+	"github.com/square/p2/pkg/util"
+)
+
+type lcBackend struct {
+	conns int
+
+	// disabling allows us to gracefully release connections without assigning
+	// new connections to the backend.
+	disabled bool
```

Comment:
Two suggestions:

1. Flip this variable to be `enabled bool`
2. Change `LeastConnections.backends` below to be a pointer-less `map[string]lcBackend`

Then the struct's zero value corresponds to a disabled host with no connections, and you can rely on that zero value to simplify the logic when reading from `backends`.